tests/integration.rs

```diff
@@ -19,6 +19,339 @@ use core::time::Duration;
+        valid_demo_root_2,
+        valid_demo_root_3,
+        valid_demo_root_4,
+    ] {
+        let cert = CertificateDer::from(cert);
+        let cert = webpki::EndEntityCert::try_from(&cert).unwrap();
+
+        for &crls in &[
+            None,
+            Some(&roots_crls),
+            Some(&intermediates_crls),
+            Some(&all_crls),
+        ] {
+            assert!(
+                cert.verify_for_usage(
+                    webpki::ALL_VERIFICATION_ALGS,
+                    &anchors,
+                    &intermediates,
+                    time,
+                    KeyUsage::server_auth(),
+                    crls.map(|l| revocation_options_for_test(l)),
+                    None,
+                )
+                .is_ok()
+            );
+
+            assert!(
+                cert.verify_for_usage(
+                    webpki::ALL_VERIFICATION_ALGS,
+                    &legacy_anchors,
+                    &intermediates_legacy,
+                    time,
+                    KeyUsage::server_auth(),
+                    crls.map(|l| revocation_options_for_test(l)),
+                    None,
+                )
+                .is_ok()
+            );
+
+            let path = cert
+                .verify_for_usage(
+                    webpki::ALL_VERIFICATION_ALGS,
+                    &all_anchors,
+                    &intermediates_legacy,
+                    time,
+                    KeyUsage::server_auth(),
+                    crls.map(|l| revocation_options_for_test(l)),
+                    None,
+                )
+                .unwrap();
+
+            // verify should find shortest path
+            assert!(anchors.contains(path.anchor()));
+        }
+    }
+
+    for &(cert, _dns_name) in &[
+        revoked_demo_root_1,
+        revoked_demo_root_2,
+        revoked_demo_root_3,
+        revoked_demo_root_4,
+    ] {
+        let cert = CertificateDer::from(cert);
+        let cert = webpki::EndEntityCert::try_from(&cert).unwrap();
+
+        for &crls in &[None, Some(&roots_crls)] {
+            assert!(
+                cert.verify_for_usage(
+                    webpki::ALL_VERIFICATION_ALGS,
+                    &anchors,
+                    &intermediates,
+                    time,
+                    KeyUsage::server_auth(),
+                    crls.map(|l| revocation_options_for_test(l)),
+                    None,
+                )
+                .is_ok()
+            );
+        }
+
+        for &crls in &[&intermediates_crls, &all_crls] {
+            assert!(
+                cert.verify_for_usage(
+                    webpki::ALL_VERIFICATION_ALGS,
+                    &anchors,
+                    &intermediates,
+                    time,
+                    KeyUsage::server_auth(),
+                    Some(revocation_options_for_test(crls)),
+                    None,
+                )
+                .is_err_and(|e| matches!(e, webpki::Error::CertRevoked))
```

Comment:
@cpu @djc to the best of my ability, I couldn’t see any other test cases where revocation checking occurs,
1. with real-world CRLs (as opposed to those constructed by generate.py), i.e. production data and not trivially small data (hypothesis being that real data can be weird and thus useful for finding edge cases and regressions)
2. where the collection of CRLs are from multiple authorities

Reply:
The assertion immediately above this one is complementary (shows the revoked certificates will verify if the crls from the intermediate CAs are not given).